docs/core/development/dev-process.rst

```diff
@@ -0,0 +1,106 @@
+Development process
+===================
+
+This page described the process that should be followed for contributing to Twisted.
+
+
+If you're going to be doing development on Twisted itself,
+or if you want to take advantage of bleeding-edge features (or bug fixes) that are not yet available in a numbered release,
+you'll probably want to check out a tree from the Twisted Git repository.
+
+The `trunk` is the main branch and is where all current development takes place.
+
+
+Checkout and initial dev virtual environment
+--------------------------------------------
+
+Twisted used Git to track code changes.
+Git tutorials can be found elsewhere,
+see in particular `Git and GitHub learning resources <https://help.github.com/articles/good-resources-for-learning-git-and-github/>`_ :
+
+.. code-block:: console
+
+    $ git clone https://github.com/twisted/twisted twisted
+    $ virtualenv venv
+    $ . venv/bin/activate
+    $ pip install -e .[dev]
+    $ pip install pre-commit
+
+The output of ``git blame`` `will be better <https://github.com/psf/black#migrating-your-code-style-without-ruining-git-blame>`_ if you configure it to use our ignore file:
+
+.. code-block:: console
+
+    $ cd twisted
+    $ git config blame.ignoreRevsFile .git-blame-ignore-revs
+
+
+Running tests
+-------------
+
+The tests are handled by the `trial` testing framework.
+It knows about deferred and any other good stuff provided by Twisted.
+To run the full unit-test suite, do:
+
+.. code-block:: console
+
+    ./bin/trial twisted
```

Comment:
In modern Python environments, we should be encouraging users to use the entrpoints-defined scripts rather than the things in `./bin`.

(For that matter... I think `./bin` has outlived its usefulness entirely? Perhaps time to get rid of it. But that's for a different PR.)

```suggestion
    trial twisted
```